pairing/bn254/adapter.go

```diff
@@ -0,0 +1,50 @@
+package bn254
+
+import (
+	"github.com/drand/kyber"
+)
+
+// SuiteBn254 is an adapter that implements the suites.Suite interface so that
+// bn254 can be used as a common suite to generate key pairs for instance but
+// still preserves the properties of the pairing (e.g. the Pair function).
+//
+// It's important to note that the Point function will generate a point
+// compatible with public keys only (group G2) where the signature must be
+// used as a point from the group G1.
```

Comment:
This is part of the code I spliced from the existing `bn256` package. I guess it's because the `Point` function specifically generates a point on G2. Also, I did not implement a hash-to-point function for G2, which would be needed to hash messages to G2 for signing.